lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -618,7 +618,7 @@ def add_subscription(xml, options, reference = nil)
 
         xml.tag! 'recurringSubscriptionInfo' do
           if reference
-            _, subscription_id, _ = reference.split(";")
+            subscription_id = reference.split(";")[-1]
```

Comment:
Hrrm. This will break if we have to add yet another thing to the end of the auth. A a positive reference might be better (`[6]`).

Reply:
This doesn't work because not all of the options are passed in every time. The reference can have a variable number of objects.

Reply:
Are you sure? The semicolons should still be added.

Oh I see why, the auth string is being `compact`ed first. https://github.com/deedeelavinder/active_merchant/blob/8afd6a18a2118b85a871e56c534031c258a6ac72/lib/active_merchant/billing/gateways/cyber_source.rb#L712
That should probably not be done.  It doesn't look like that would impact any of the other `split`s, but it is worth double checking.

It might be nice to pull out it out into its own `authorization_from` method like other gateways, too.

Reply:
@curiousepic 👍  Added an `authorization_from` method with no `compact`.